### DISCUSSION TITLE
Would love some input regarding linker issue

### DISCUSSION THREAD

**Discussion by 573** [nix-community/nix-on-droid#170] [General]
I tried to build a haskell project in nix-on-droid.

The linker fails trying to access `/lib` though.

I created a ticket over here https://github.com/NixOS/nixpkgs/issues/155722

In the ticket there is also a repository linked (https://github.com/573/aarch64-linux-ld-issue155722-reproduce/runs/5044168375?check_suite_focus=true#step:6:336 , sadly build run in continuous integration currently is blocked) where I try to reproduce the error.

Is there any chance the `/lib` issue might be related to how nix-on-droid / termux works in general ?

**Comment by t184256** [comment]
Well, I guess in Nix-on-Droid `/lib` exists, but is unreadable, which is a rather exotic/untested scenario that doesn't crop up with regular Nix-on-otherLinuxes.

To try and rule out ARM strangeness, you could also try building on a different-arch NixOS box with QEMU user emulation: `boot.binfmt.emulatedSystems = [ "aarch64-linux" ];`.

**Reply by 573** [reply]
@t184256 thanks for the input.

this was like a first real test case for my remote builder learning project.

indeed with the remote builder, binfmt-emulation approach you suggested, the linker went through like a charm.

the rest i. e. why `/lib` can't be substituted when not accessible / existent, remains a black box for now.

**Reply by t184256** [reply]
\*sigh* somewhere, a dozen abstraction layers below, there's gonna be a `if exists(dir): list(dir)` instead of a `if readable(dir): list(dir)`, which is technically a bug, but has never cropped up before because who would have a libdir that's not readable?
